### PR TITLE
Create unified CSS Grid utility, refactor the majority of .row, .col, .grid-row, and .grid-col code

### LINKED ISSUE #267: Fix .row, .col, .grid-row, and .grid-col
This is possibly one of the worst artifacts I've ever introduced into the codebase: we have two pairs of classes,`.row` /`.col` and `.grid-row`/ `.grid-col`, and they're both *really bad*.

As one of my last actions in the dev team ... let's fix that. I will:

* remove **every instance** of all of these classes
* if possible, lay out all dependent classes using fixed-template CSS Grid
* evaluating if needed, I will also write a clear, reusable flexbox CSS option

### PR DESCRIPTION
**Pull request by mattxwang**
Wow, this is a huge PR. This tackles #267 in most senses, but leaves in a stopgap in case I've made a mistake.

The biggest change is that I've made a new file, `_grid.scss`, that encompasses a grid generator. You can pass in a class name, grid ratio, and breakpoint, and it will automatically generate a set of CSS Grid classes that fulfill these parameters (currently proper "fake" column gutters with padding and working reversal for 2-column grids).

I have then used this new class to replace almost every two-column and three-column layout on the site. This includes:

* the home page
* the dev page - hero layout and projects grid
* the classes main page
* the classes grids for AI/ML and React Native (Python doesn't use the unified classes css)
* the people grids on "team" for board and general members - this does not touch the inner "person" card code
* the partners page, which i also rename to `partners.html` and standardize the layout for
* the join us page 
* all the events pages (AI Ethics, Careers, Citylab) - this enabled me to delete `_events.scss`
* the docs page (which got a minor redesign as well, as there were some extraneous `.row` and `.col` classes)
* removed them from the resources page

In general, this change (along with the creation of a few CSS utility classes and a SASS function) allowed me to remove a significant amount of duplicated/confusing CSS. Hopefully, this makes life for future developers easier!

Things that I (or someone else) will tackle in another PR:

* row-reverse capability in a grid size larger than 2 - we never use this, so I didn't implement it
* resolving the remaining team/author-related uses of `.grid-row` and `.grid-col` - may be tackled by Jun in his team page redesign. I marked these cases with `.TEAM-SPECIAL`, so they should be obvious to spot.

I'm pretty tired, so I may have missed something. I think this is going to require a thorough review or two to make sure that I didn't break any of the pages on the site.

Mostly closes #267.

### CODE REVIEW

File: _sass/_grid.scss [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    .row#{$grid-prefix}-ratio-#{$grid-name} {
+      display: grid;
+      grid-template-columns: $grid-ratio;
+
+      .col-padded {
+        padding-left: 1em;
+        padding-right: 1em;
+      }
+
+      .col-padded:nth-child(#{$column-size}n - 1) {
```
Comment (surreally) [inline _sass/_grid.scss:34]: should probably be `#{$column-size}n`. items are indexed from 1 and the last cell in each row should simply be indexed at multiples of `$column-size`. e.g. on [this page](https://deploy-preview-268--acm-teach-la-website.netlify.app/citylab-cyber), passworks has both left and right padding, which might not be desirable

Comment (mattxwang) [reply _sass/_grid.scss]: Great catch! Resolved!

File: join.html [hunk 2]
```diff
@@ -127,17 +132,14 @@ <h2 class="title">Dev</h2>
...
-        <div class="col">
+    <div class="row-desktop-ratio-1-1">
+        <div class="col-padded">
             <img class="img-responsive" src="{{site.baseurl}}/img/undraw/join-events.svg" alt="Man pointing to events on a calendar"/>
         </div>
-        <div class="col no-gutter">
+        <div class="col-padded
             <p>
                 Our special events team hosts lots of <b>stand-alone</b> events and <b>short courses</b>, usually involving <b>collabs with other orgs</b>. We aim to <b>introduce CS</b> to kids and show them what it’s all about! <a class="tla-link" href="{{site.baseurl}}/citylab-cyber">Cyber Day with CityLab</a> is an example of a past event that we’ve had. Interested? Here’s what we do:
             </p>
```
Comment (krashanoff) [inline join.html:145]:
```suggestion
        <div class="col-padded">
            <p>
                Our special events team hosts lots of <b>stand-alone</b> events and <b>short courses</b>, usually involving <b>collabs with other orgs</b>. We aim to <b>introduce CS</b> to kids and show them what it’s all about! <a class="tla-link" href="{{site.baseurl}}/citylab-cyber">Cyber Day with CityLab</a> is an example of a past event that we’ve had. Interested? Here’s what we do:
            </p>
```

**Review by krashanoff — CHANGES_REQUESTED** [review]
Phew. Sorry for the delay on review, this was a chunky one. Matt, you're absolutely nuts for doing this all.

Things look good to merge following Jun's catch and correcting small typo. Aside from that I didn't see anything out of place. Reversing more than two columns really isn't a huge concern at the moment, though it shouldn't be too tricky to implement in another PR. Overall, this is a *really* clean fix to standardizing the static site's layout. Thanks Matt!

Also thanks for fixing use of `<h1>` where it shouldn't have been, and for removing all the trailing whitespace in our source. There was ...a lot.

File: join.html (hunk 2 above)
Comment (mattxwang) [reply join.html:145]: Thanks for the catch here!

**Comment by mattxwang** [comment]
Oh @surreally do you mind if we move your change to a different branch? Don't want to make this PR too chonky haha!

**Comment by surreally** [comment]
> Oh @surreally do you mind if we move your change to a different branch? Don't want to make this PR too chonky 👀

Yeah.

**Review by krashanoff — APPROVED** [review]
LGTM, feel free to rebase or force push this branch prior to merge to resolve the overlapping work.

**Comment by mattxwang** [comment]
@surreally let me know when you've cherry-picked/otherwise prepped to open your PR, I'm going to delete this branch as soon as I merge!

**Comment by surreally** [comment]
@mattxwang I saved a patch.

**Pull request merged**